src/class-autoloader.php

```diff
@@ -63,20 +63,21 @@ public function __invoke( string $classname ) {
 			return;
 		}
 
-		// Remove the namespace.
-		$classname = \preg_replace( '#^' . \preg_quote( $this->namespace, '#' ) . '#', '', $classname );
-		$classname = \ltrim( $classname, '\\' );
+		// Break up the classname into parts.
+		$parts = \explode( '\\', $classname );
 
-		// Convert to segments.
-		$classname = \strtolower( $classname );
-		$classname = \str_replace( [ '\\', '_' ], [ '/', '-' ], $classname );
-		$classes   = \explode( '/', $classname );
+		// Retrieve the class name (last item) and convert it to a filename.
+		$class = \strtolower( \str_replace( [ '\\', '_' ], [ '/', '-' ], \array_pop( $parts ) ) );
```

Comment:
there won't be any `\\` right, since we exploded on it?
```suggestion
		$class = \strtolower( \str_replace( '_', '-', \array_pop( $parts ) ) );
```